File: frontend/src/app/dto/Comment.ts
import { Post } from "./Post";
import { User } from "./User";

export class Comment {

  id: number | null = null;
  body: string | null = null;
  user: User | null = null;
  post: Post | null = null;
  createdDateTime: string | null = null;  // `yyyy-MM-dd HH:mm:ss

  constructor(o?: Partial<Comment>) {
    Object.assign(this, o);
  }
}

File: backend/src/main/java/com/example/backend/entity/Comment.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Comment extends AuditLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String body;

    // COMMENT MANY TO ONE USER INFO
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userInfo_id")
    private UserInfo userInfo;

    // COMMENT MANY TO ONE POST
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    private Post post;

    public Comment(Long id) {
        this.id = id;
    }
}
